### DISCUSSION TITLE
`Rebuild`

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#30] [Q&A] [answered]
I have this existential doubt: If I'm missing a `rebuild()` call who will miss the update:
a) Only capsules previously using this capsule.
b) All capsules, including the ones that begin to use this capsule after the missing `rebuild()`. (Meaning that they will get a previous state)

(**) I mean missing a `rebuild()` call after modifying a set/map created with `use.value()`

**Comment by GregoryConrad** [comment, marked as answer]
﻿﻿﻿Rebuilds will rebuild the capsule’s data, and the container will cache the new data. Thus, any downstream capsules at the time of the rebuild or in the future would be affected.

_But you often don’t need to worry about this._ If you’re only using the built in side effects, all rebuilds will be managed for you.

**Reply by GregoryConrad** [reply]
Ahh, and as far as use.value, it depends on whether or not you are returning any of the data within the use.value out. If you are only returning functions that operate on use.value, no need to rebuild. If you are returning the value itself (or some transformation of it), then you may want to rebuild (but manually controlling rebuilds like this is discouraged, use immutability instead).
